src/lib/search-params.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  SearchOptions,
+  SortBy as SortBySearchOption,
+  SortOrder,
+} from '@/lib/dataset-fetcher';
+
+export enum SortBy {
+  RelevanceDesc = 'relevanceDesc',
+  NameAsc = 'nameAsc',
+  NameDesc = 'nameDesc',
+}
+
+export const defaultSortBy = SortBy.RelevanceDesc;
+
+const sortMapping = {
+  [SortBy.RelevanceDesc]: {
+    sortBy: SortBySearchOption.Relevance,
+    sortOrder: SortOrder.Descending,
+  },
+  [SortBy.NameAsc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Ascending,
+  },
+  [SortBy.NameDesc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Descending,
+  },
+};
+
+interface ToSearchParamsProps {
+  query?: SearchOptions['query'];
+  offset?: SearchOptions['offset'];
+  filters: {
+    licenses?: string[];
+    publishers?: string[];
+  };
+  sortBy?: SortBy;
+}
+
+export function getUrlWithSearchParams({
+  query,
+  offset,
+  filters: {licenses, publishers},
+  sortBy,
+}: ToSearchParamsProps) {
+  const searchParams: {[key: string]: string} = {};
+
+  if (query) {
+    searchParams.query = query;
+  }
+
+  if (licenses?.length) {
+    searchParams.licenses = licenses.join(',');
+  }
+
+  if (publishers?.length) {
+    searchParams.publishers = publishers.join(',');
+  }
+
+  if (offset) {
+    searchParams.offset = `${offset}`;
+  }
+
+  if (sortBy && sortBy !== defaultSortBy) {
+    searchParams.sortBy = sortBy;
+  }
+
+  const encodedSearchParams = new URLSearchParams(searchParams).toString();
+
+  if (encodedSearchParams) {
+    return '/?' + encodedSearchParams;
+  } else {
+    return '/' + encodedSearchParams;
+  }
+}
+
+interface FromSearchParamsToSearchOptionsProps {
+  publishers?: string;
+  licenses?: string;
+  query?: string;
+  offset?: string;
+  sortBy?: SortBy;
+}
+
+interface Response {
+  searchOptions: SearchOptions;
+  sortBy: SortBy;
+}
+
+// This function translates the search params to valid search options.
+// Next.js already separate the search query string into separate properties with string values.
+// Always return a valid SearchOptions object, even if the search params aren't correct,
+// so the application doesn't fail on invalid search params.
+export function fromSearchParamsToSearchOptions({
+  publishers,
+  licenses,
+  query,
+  offset = '0',
+  sortBy = defaultSortBy,
+}: FromSearchParamsToSearchOptionsProps): Response {
```

Comment:
Consideration: you could add a (Zod) validator here, for validating the values, setting default values and transforming the values (e.g. from strings to other types)?